src/components/segmented-control/segmented-control.tsx

```diff
@@ -31,10 +31,9 @@ export const SegmentedControl = <T extends unknown>({
   onChange
 }: PropsWithChildren<Props<T>>) => {
   const styles = useSegmentedControlStyles();
-  const translateX = useAnimationRef();
   const { layoutWidth, handleLayout } = useLayoutWidth();
-
-  const tileWidth = (layoutWidth - 2 * tileMargin) / (values.length || 1);
+  const tileWidth = ((width ?? layoutWidth) - 2 * tileMargin) / (values.length || 1);
+  const translateX = useAnimationRef(selectedIndex * tileWidth);
```

Comment:
this is unique case, use `useRef(new Animated.Value(selectedIndex * tileWidth)).current;` instead of `useAnimationRef`  
and revert `useAnimationRef` changes back

Reply:
I wouldn't be sure this is a unique case in all app. Our designer may carry a lot of 'surprises'